model/task/task.go

```diff
@@ -1268,56 +1281,38 @@ func (t *Task) UpdateHeartbeat() error {
 	)
 }
 
-// SetPriority sets the priority of the task.
-// It increases the priority of the tasks it depends on.
-func (t *Task) SetPriority(priority int64, user string) error {
-	t.Priority = priority
-	depTasks, err := GetRecursiveDependenciesUp([]Task{*t}, nil)
-	if err != nil {
-		return errors.Wrap(err, "error getting task dependencies")
-	}
+// Blacklist sets the priority of a task so it will never run.
+// It also deactivates the task and any tasks that depend on it.
+// The build cache is not updated
+func (t *Task) Blacklist(user string) ([]Task, error) {
```

Comment:
How come this function can't just call SetTaskPriority? If SetTaskPriority won't properly handle the blacklist case, then it should probably just be modified to handle it, or shouldn't accept it.

Reply:
😞 This is a good question.
I wanted to move SetTaskPriority into the model package so it could update the build cache (to [consolidate](https://github.com/evergreen-ci/evergreen/pull/3689#discussion_r442276859) the places where we update the build cache) but the commitqueue package couldn't import the model package. The commit queue can import the task and build packages, though, so I made this function to do just what the commit queue needed for the task (even though it's a subset of what  SetTaskPriority already does and is a violation of DRY). The commit queue calls another function in the build package to update the build cache.